### PR TITLE
[10.x] update HasAttributes trait by using match instead of if-elseif-else 

### PR DESCRIPTION
**Pull request by saMahmoudzadeh**
Hello!

This PR,the if-elseif-else statements in the `getCastType` and `setEnumCastableAttribute` methods have been converted to match expressions in PHP 8.0.

Whenever an if statement ultimately ends with a return, the elseif structure can be converted to an if statement. This change improves code readability. Additionally, in the `originalIsEquivalent` method, the if-elseif statements have been converted to if blocks.

Thanks!

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php [hunk 1]
```diff
@@ -862,17 +862,13 @@ protected function getCastType($key)
...
-        } elseif ($this->isImmutableCustomDateTimeCast($castType)) {
-            $convertedCastType = 'immutable_custom_datetime';
-        } elseif ($this->isDecimalCast($castType)) {
-            $convertedCastType = 'decimal';
-        } elseif (class_exists($castType)) {
-            $convertedCastType = $castType;
-        } else {
-            $convertedCastType = trim(strtolower($castType));
-        }
+        $convertedCastType = match ($castType) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php:865]: this is incorrect

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Refactors to change things to match are largely a waste of time IMO.

**Comment by saMahmoudzadeh** [comment]
> Refactors to change things to match are largely a waste of time IMO.

I appreciate the guidance. From now on, I will try to develop the features I previously noted in Laravel.

Translation: I appreciate the guidance. I will strive to work on the features I previously mentioned for Laravel development from now on.

@GrahamCampbell

**Review by saMahmoudzadeh — COMMENTED** [review]
OK, Thank you for the guidance. The changes will be applied.

**Pull request closed without merging**